posts/my-journey-with-creating-my-first-slackbot.md

```diff
@@ -0,0 +1,199 @@
+}
+catch (error) {
+  console.error(error);
+}
+```
+
+
+### The markup to [retrieve conversation history](https://api.slack.com/messaging/retrieving) from your Slack channel is as follows:
+
+``` javascript
+// Store conversation history
+let conversationHistory;
+// ID of channel you watch to fetch the history for
+let channelId = "C24601";
+
+try {
+  // Call the conversations.history method using WebClient
+  const result = await client.conversations.history({
+    channel: channelId
+  });
+
+  conversationHistory = result.messages;
+
+  // Print results
+  console.log(conversationHistory.length + " messages found in " + channelId);
+}
+catch (error) {
+  console.error(error);
+}
+```
+
+
+### The markup to [retrieve an individual post](https://api.slack.com/messaging/retrieving) from your Slack channel is as follows:
+
+``` javascript
+// Store message
+let message;
+
+// Fetch conversation history using the ID and a TS from the last example
+async function fetchMessage(id, ts) {
+  try {
+    // Call the conversations.history method using the built-in WebClient
+    const result = await app.client.conversations.history({
+      // The token you used to initialize your app
+      token: "xoxb-your-token",
+      channel: id,
+      // In a more realistic app, you may store ts data in a db
+      latest: ts,
+      // Limit results
+      inclusive: true,
+      limit: 1
+    });
+
+    // There should only be one result (stored in the zeroth index)
+    message = result.messages[0];
+    // Print message text
+    console.log(message.text);
+  }
+  catch (error) {
+    console.error(error);
+  }
+}
+
+// Fetch message using a channel ID and message TS
+fetchMessage("C12345", "15712345.001500");
+```
+
+
+## My ‘Scheduling A Future Post’ Bug Issue Encounter
+
+I was able to achieve three of the task above with my Slackbot. I was able to post a message, retrieve conversation history, retrieve an individual message from the Slack channel. However, I ran into an issue when it came to scheduling a post at a later date and time. When I attempted to schedule a post, I kept getting an error message. One of the lines of the error message, I deemed important, is below.
+
+`Error: An API error occurred: invalid_arguments`
+
+
+So, I decided to use my friend Google and see if anyone else may have had this same issue. I was able to locate a small hand full of websites of others who had the same issue but, their solutions didn’t quite work for me. The unusual thing about the issue is the fact that I copied the exact documentation code snippet from the website and placed it in my code, and it still didn’t work.
+
+During my online research, I read an article of someone that appeared to be a Slack API employee or partner that said they are aware of the issue and will update the documentation on their website. However, that didn’t help me at the time, therefore, I concluded, I wasn’t going to get anywhere by just googling the error message.
```

Comment:
```suggestion
During my online research, I read an article of someone that was a Slack API employee or partner that said they are aware of the issue and will update the documentation on their website. However, that didn’t help me at the time, therefore, I concluded, I wasn’t going to get anywhere by just googling the error message.
```

Reply:
I chose to use the phrase "appeared to be" because I wasn't 100% sure the person was a Slack API employee